### ISSUE TITLE
[Accessibility] Narrator cannot announce order information of the sub menu items in MenuStrip

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#3471]
* .NET Core Version: 
.Net 5.0.100-preview.7.20314.1 from Net 5.0 master branch

* Have you experienced this same bug with .NET Framework?: 
Yes

**Minimal repro:**
1. Extract WinformsFormsApp6.zip with MenuStrip and open it in VS, then build and run this application.
[WindowsFormsApp6.zip](https://github.com/dotnet/winforms/files/4803525/WindowsFormsApp6.zip)
2. Start Narrator, and focus on File menu, then expand it, go through its sub-items “New, Open, Save, Save As, Print, Print Preview, Exist” one by one

**Actual behavior:**
Narrator cannot announce order information of the sub menu items in MenuStrip:
![image](https://user-images.githubusercontent.com/26474449/85115411-32eff000-b24e-11ea-871d-5309f4e5d069.png)

**Expected behavior:**
Narrator can announce order information of the sub menu items in MenuStrip

**Comment by merriemcgaw** [comment]
@Olina-Zhang are you referring to the position of the menu item in the list (7 of 7)?

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang are you referring to the position of the menu item in the list (7 of 7)?

@merriemcgaw, Yes, narrator cannot read it, like this: 'which one' of 'total count'.

**Comment by merriemcgaw** [comment]
I'll circle back with the CAIAcc experts and find out if these announcements are nice to have or required.

**Comment by tinodo** [comment]
You can override the AccessibilityRole in code. Make the Menu a List and the items ListItems. This will cause Narrator to announce the item count and position of an item.

**Comment by merriemcgaw** [comment]
I think this is a "nice to have" and we can consider it for .NET 8. I don't think it meets the bar for coming in hot in .NET 7. We haven't gotten any official bugs about this behavior so I think we're good to handle this on in our own time.

**Comment by Nora-Zhou01** [comment]
Verified with .NET SDK 8.0.100-alpha.1.23068.16 build from main branch, this issue is fixed that Narrator can announce order information of the sub menu items in MenuStrip.
![n1](https://user-images.githubusercontent.com/104609169/213378437-664d2f69-2233-47ab-a703-b3494be04d7c.png)

**Comment by merriemcgaw** [comment]
@Olina-Zhang can you make sure that NVDA and JAWS are both working correctly with this fix?

**Comment by Nora-Zhou01** [comment]
Verified with .NET SDK 8.0.100-alpha.1.23068.16 build from main branch,NVDA can announce order information of the sub menu items in MenuStrip,but JAWS cannot announce order information of the sub menu items in MenuStrip.
NVDA:
![N](https://user-images.githubusercontent.com/104609169/213605931-00e28be1-5e5f-4915-959b-4d4c9526b3f3.png)

**Comment by Cassie-Li01** [comment]
Verified on 8.0 TP version .NET 8.0.100-preview.1.23105.3, issue is fixed. Screenshot same as above.

**Comment by Cassie-Li01** [comment]
Verified with .NET 8 Preview 1 TP build: .NET 8.0.100-preview.1.23112.5, issue is fixed. Screenshot as above
